examples/config.example.json

```diff
@@ -20,5 +20,26 @@
       "schema_files": {
             "submission": "bouncerscript/data/bouncer_submission_task_schema.json",
             "aliases": "bouncerscript/data/bouncer_aliases_task_schema.json"
+      },
+
+      "aliases_regexes": {
+             "thunderbird-beta-latest": "^Thunderbird-\\d+\\.0b\\d+$",
```

Comment:
I agree: in the future we probably want to support `script_config.yml`; this will hopefully be a lot less ugly there.

Reply:
I'm worried transposing this config in puppet may bite us one day. The regex is written in Puppet language to be transposed in a JSON file thanks to ERB. This JSON is then interpreted to become a python regex. Simply put:

```
Puppet => ERB => JSON => Python
```

I don't see why a bouncer instance would ever define something different for `"thunderbird-beta-latest"` (unless a bug is introduced). Although, I see why a Thunderbird bouncer instance wouldn't bother to define Firefox config: smaller config + unable to modify Firefox stuff.

Thus, in order to achieve these goals, I would recommend to:

* define all regexes for all supported products in a file like `constants.py` (à la beetmover). This would avoid to pass regex strings through 4 different languages
* implement a second check to make sure we don't modify Firefox aliases when a Thunderbird payload is passed.

What do you think?

Reply:
To begin with, I agree that having this translate via four languages sucks. I was hoping we move towards `yml` as @escapewindow suggested soon so this could be flushed away. 

Reasons behind moving this under puppet were:
1. Easier to push `puppet fix` + `rerun task` if there's an issue with the regexes or a new alias is added. Keeping these under `constants.py` would mean we need to spin another `bouncerscript version` + `puppet version bump` + `rerun task`. So I guess one step shorter. Not much, but had that in mind.

2. Security wise - as you mentioned, it's easier to separate the waters from TB / Firefox directly from the puppet configs. That way we ensure the future TB bouncer-dev instance shall have a regexes narrowed to thunderbird entries and nothing more, while the Firefox counterpart shall contain the rest. 

3. Security. Having some of these checks in puppet meant in my head more security than betting all the money on the bouncerscript code. If someone hacks that and munges the checks, things can go south. But as I think about it, this is non-sense, as you need access to puppet anyway to push a new bouncerscript version. If one grabbed that, it can munge the puppet code directly there so it's the same thing. So yeah, I was wrong from this perspective. 

Happy to move this under `constants.py` instead but I'd need to follow-up with some in-tree patches as well to make sure product is passed down the payload.

Reply:
In my opinion, yaml would just shift the problem away, it just replaces json by another type of language.

I agree with 1. It's also easier to footgun with a bad puppet patch. Having this in constants.py allows us to have unit test, and thus ensure the fix is good at PR time.

Regarding 2, scriptworker for instance, defines some thunderbird specific config: https://github.com/mozilla-releng/scriptworker/blob/523eafed7772dff6815f9a98ae6bfb774c1896b5/scriptworker/constants.py#L209 

You can detect the product by looking at the scopes. As of now, thunderbird scopes start with `project:comm:thunderbird:releng:`. See [Bug 1421062](https://bugzilla.mozilla.org/show_bug.cgi?id=1421062). At the moment, I'd say there's no need to support Thunderbird yet. We can just assume the product is Firefox, until a bug like 1421062 takes care of it. What do you think?

Reply:
1. Yeah, I agree with the argument on better unittesting the python code versus puppet. Let's do that
2. Sounds good, I'll file an issue here just so that we don't forget to do this when the time comes in TB.

Thanks for your comments again.

Reply:
I filed https://github.com/mozilla-releng/bouncerscript/issues/18 to continue this conversation when the time comes.